### PR TITLE
Run GCE Authenticator tests in Jenkins

### PR DESCRIPTION
**Pull request by orenbm**
Run GCE Authenticator tests in Jenkins

### CODE REVIEW

File: cucumber/authenticators_gce/features/authn_gce_hosts.feature [hunk 1]
```diff
@@ -107,11 +76,9 @@ Feature: GCP Authenticator - Test hosts can authentication scenarios
...
 
   Scenario: Host with all valid annotations and an illegal annotation key is denied
-    Given I have host "illegal-annotation-key-test-app"
-    And I grant group "conjur/authn-gce/apps" to host "illegal-annotation-key-test-app"
-    And I obtain a GCE identity token in full format with audience claim value: "conjur/cucumber/host/illegal-annotation-key-test-app"
-    And I set all valid GCE annotations to host "illegal-annotation-key-test-app"
-    And I set "authn-gce/invalid-key" annotation to host "illegal-annotation-key-test-app"
+    Given I set "authn-gce/invalid-key" annotation to host "test-app"
+    And I set all valid GCE annotations to host "test-app"
+    And I set "authn-gce/invalid-key" annotation to host "test-app"
```
Comment (eranha) [inline cucumber/authenticators_gce/features/authn_gce_hosts.feature:81]: Duplicate my bad.

**Review by eranha — CHANGES_REQUESTED** [review]
Well done.

File: dev/cli [hunk 2]
```diff
@@ -37,10 +37,19 @@ exit
 
 function enable_gce() {
   echo "Setting GCE details as env variables"
-  local gce_env_args="-e GCE_INSTANCE_IP=$GCE_INSTANCE_IP"
-  gce_env_args="$gce_env_args -e GCE_INSTANCE_NAME=$GCE_INSTANCE_NAME"
-  gce_env_args="$gce_env_args -e GCE_INSTANCE_USERNAME=$GCE_INSTANCE_USERNAME"
-  gce_env_args="$gce_env_args -e GCE_PRIVATE_KEY_PATH=$GCE_PRIVATE_KEY_PATH"
+  # read token and set env vars
+  gce_token=$(cat authn-gce/tokens/gce_token_valid)
```
Comment (eranha) [inline dev/cli:41]: Check if token file exists.

**Review by eranha — CHANGES_REQUESTED** [review]

**Review by nessiLahav — APPROVED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -97,6 +97,59 @@ pipeline {
...
+                    local filename="$3"
+
+                    curl \
+                      -s \
+                      -H 'Metadata-Flavor: Google' \
+                      "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?format=${token_format}&audience=${audience}" \
+                      > $filename
+                  }
+
+                  get_token_into_file "full" "conjur/cucumber/host/test-app" "gce_token_valid"
```
Comment (eranha) [inline Jenkinsfile:123]: The / in audience deos not require URL encoding?

Comment (orenbm) [reply Jenkinsfile:123]: no

**Pull request merged**